app/models/user.rb

```diff
@@ -400,6 +400,14 @@ def has_devices?
     !recent_devices.empty?
   end
 
+  def new_device(cookies)
```

Comment:
* Since this method returns a boolean, I'd [conventionally](https://rubystyle.guide/#bool-methods-qmark) expect it to end in `?`
* I don't feel _too_ strongly about it, but my personal rule of thumb with positional vs. keyword argument is whether it'd be obvious by the method name alone what it expects as arguments; I don't think it is obvious here, so I might opt for a keyword argument.
* Do we need to send the full cookies object, or maybe we can just send the device cookie?

```suggestion
  def new_device?(device_cookie:)
```

Reply:
That's updated to flag itself as boolean and use keyword argument.